### PR TITLE
Il 547 add metadata to overviews

### PR DESCRIPTION
**Pull request by MerlinKallenbornAA**
# Description
No description.

## Before Merging
 - [ ] Review the code changes
    - Unused print / comments / TODOs
    - Missing docstrings for functions that should have them
    - Consistent variable names
    - ...
 - [ ] Update `changelog.md` if necessary
 - [ ] Commit messages should contain a semantic [label](https://gist.github.com/joshbuchea/6f47e86d2510bce28f8e7f42ae84c716) and the ticket number
   - Consider squashing if this is not the case

### CODE REVIEW

File: src/documentation/parameter_optimization.ipynb [hunk 1]
```diff
@@ -306,7 +311,7 @@
    "name": "python",
    "nbconvert_exporter": "python",
    "pygments_lexer": "ipython3",
-   "version": "3.12.2"
+   "version": "3.11.8"
```
[the remark this answers is not in the document]
Comment (SebastianNiehusAA) [reply src/documentation/parameter_optimization.ipynb:314]: I also talked with Merlin about it. It's just the metadata and not really relevant

File: src/intelligence_layer/evaluation/evaluation/evaluator/evaluator.py [hunk 2]
```diff
@@ -118,12 +123,18 @@ def evaluate_runs(
             abort_on_error: Flag to abort all evaluations when an error occurs. Defaults to False.
             skip_example_on_any_failure: Flag to skip evaluation on any example for which at least one run fails. Defaults to True.
             description: Optional description of the evaluation. Defaults to None.
+            labels: A list of labels for filtering. Defaults to an empty list.
+            metadata: A dict for additional information about the evaluation overview. Default to an empty dict.
```
Comment (SebastianNiehusAA) [inline src/intelligence_layer/evaluation/evaluation/evaluator/evaluator.py:127]:
```suggestion
            metadata: A dict for additional information about the evaluation overview. Defaults to an empty dict.
```

File: src/intelligence_layer/evaluation/evaluation/evaluator/incremental_evaluator.py [hunk 3]
```diff
@@ -141,12 +146,18 @@ def evaluate_additional_runs(
             num_examples: The number of examples which should be evaluated from the given runs.
                 Always the first n runs stored in the evaluation repository. Defaults to None.
             abort_on_error: Flag to abort all evaluations when an error occurs. Defaults to False.
+            labels: A list of labels for filtering. Defaults to an empty list.
+            metadata: A dict for additional information about the evaluation overview. Default to an empty dict.
```
Comment (SebastianNiehusAA) [inline src/intelligence_layer/evaluation/evaluation/evaluator/incremental_evaluator.py:150]:
```suggestion
            metadata: A dict for additional information about the evaluation overview. Defaults to an empty dict.
```

File: src/intelligence_layer/evaluation/run/runner.py [hunk 4]
```diff
@@ -97,11 +102,17 @@ def run_dataset(
             max_workers: Number of examples that can be evaluated concurrently. Defaults to 10.
             description: An optional description of the run. Defaults to None.
             trace_examples_individually: Flag to create individual tracers for each example. Defaults to True.
+            labels: A list of labels for filtering. Defaults to an empty list.
+            metadata: A dict for additional information about the run overview. Default to an empty dict.
```
Comment (SebastianNiehusAA) [inline src/intelligence_layer/evaluation/run/runner.py:106]:
```suggestion
            metadata: A dict for additional information about the run overview. Defaults to an empty dict.
```

**Review by SebastianNiehusAA — CHANGES_REQUESTED** [review]

File: src/intelligence_layer/evaluation/aggregation/domain.py [hunk 5]
```diff
@@ -41,6 +47,8 @@ class AggregationOverview(BaseModel, Generic[AggregatedEvaluation], frozen=True)
     crashed_during_evaluation_count: int
     description: str
     statistics: SerializeAsAny[AggregatedEvaluation]
+    labels: set[str] = set()
```
Comment (FelixFehse) [inline src/intelligence_layer/evaluation/aggregation/domain.py:50]: are we sure that e.g. the `set` is created newly for each instance, or is it shared between instances and leads to unexpected behavior?

File: src/intelligence_layer/evaluation/dataset/domain.py [hunk 6]
```diff
@@ -60,13 +62,22 @@ class Dataset(BaseModel):
     Attributes:
         id: Dataset ID.
         name: A short name of the dataset.
+        label: Labels for filtering datasets. Defaults to empty list.
+        metadata: Additional information about the dataset. Defaults to empty dict.
     """
 
     id: str = Field(default_factory=lambda: str(uuid4()))
     name: str
+    labels: set[str] = set()
```
Comment (FelixFehse) [inline src/intelligence_layer/evaluation/dataset/domain.py:71]: [see other comment regarding default values]

File: src/intelligence_layer/evaluation/aggregation/domain.py (hunk 5 above)
Comment (SebastianNiehusAA) [reply src/intelligence_layer/evaluation/aggregation/domain.py:50]: They are not shared and the tests cover this

File: src/intelligence_layer/evaluation/dataset/domain.py (hunk 6 above)
Comment (SebastianNiehusAA) [reply src/intelligence_layer/evaluation/dataset/domain.py:71]: [see other comment regarding default values] ;)

**Review by SebastianNiehusAA — APPROVED** [review]

**Pull request merged**